### PR TITLE
ooklaserver: init at 2.11.1.2

### PR DESCRIPTION
**Pull request by NeverBehave**
## Description of changes

Adding OoklaServer: TCP based server daemon that provides standalone testing, from speedtest.net

OoklaServer version 2.11 (2.11.1.2) build number 2024-02-13.1456.91c4f93

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/oo/ooklaserver/package.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    "i686-windows" = "OoklaServer-windows32.zip";
+    "x86_64-windows" = "OoklaServer-windows64.zip";
+    # OoklaServer-linux64-deb9.zip
+    # OoklaServer-freebsd12_64.zip
+    # OoklaServer-freebsd13_64.zip
+  };
+in
+stdenv.mkDerivation rec {
+  version = "2.11.1.2";
+  pname = "OoklaServer";
```
Comment (NickCao) [inline pkgs/by-name/oo/ooklaserver/package.nix:23]: The `pname` attribute may not contain uppercase letters.

File: pkgs/by-name/oo/ooklaserver/package.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  };
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [ unzip autoPatchelfHook ];
+
+  unpackPhase = ''
+    tar -xvf $src
+    unzip ${filenameMapping.${stdenv.system}}
```
Comment (NickCao) [inline pkgs/by-name/oo/ooklaserver/package.nix:37]: Shall we move this to `installPhase` instead?

File: pkgs/by-name/oo/ooklaserver/package.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm555 OoklaServer "$out/bin/OoklaServer"
+    cp OoklaServer.properties.default "$out"
```
Comment (NickCao) [inline pkgs/by-name/oo/ooklaserver/package.nix:43]: Please somewhat adhere to the FHS, e.g. `/etc/OoklaServer.properties.default`

File: pkgs/by-name/oo/ooklaserver/package.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (NickCao) [inline pkgs/by-name/oo/ooklaserver/package.nix:47]: Don't use `with lib;`

File: pkgs/by-name/oo/ooklaserver/package.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+}:
+let
+  filenameMapping = {
+    "x86_64-linux" = "OoklaServer-linux-x86_64-static-musl.zip";
+    # Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
+    "x86_64-darwin" = "OoklaServer-macosx.zip";
+    "aarch64-darwin" = "OoklaServer-macosx.zip";
+    "aarch64-linux" = "OoklaServer-linux-aarch64-static-musl.zip";
+    "i686-windows" = "OoklaServer-windows32.zip";
+    "x86_64-windows" = "OoklaServer-windows64.zip";
```
Comment (NickCao) [inline pkgs/by-name/oo/ooklaserver/package.nix:15]: I believe the binary name for windows platform is different?

File: pkgs/by-name/oo/ooklaserver/package.nix (hunk 2 above)
Comment (NeverBehave) [reply pkgs/by-name/oo/ooklaserver/package.nix:37]: It is more like unpack the right version, but I am open to that

File: pkgs/by-name/oo/ooklaserver/package.nix (hunk 5 above)
Comment (NeverBehave) [reply pkgs/by-name/oo/ooklaserver/package.nix:15]: ah good catch, it ends in `.exe`.

File: pkgs/by-name/oo/ooklaserver/package.nix (hunk 1 above)
Comment (NeverBehave) [reply pkgs/by-name/oo/ooklaserver/package.nix:23]: I do see some pname with uppercase letters, is there a standard for this?

Comment (NickCao) [reply pkgs/by-name/oo/ooklaserver/package.nix:23]: See `pkgs/README.md`, we are in the process of normalizing these.

File: pkgs/by-name/oo/ooklaserver/package.nix (hunk 5 above)
Comment (NeverBehave) [reply pkgs/by-name/oo/ooklaserver/package.nix:15]: I saw some code refer to `stdenv.isWindows` but it seems that does not exists. Any suggestion on alternative?

Comment (NickCao) [reply pkgs/by-name/oo/ooklaserver/package.nix:15]: There is `stdenv.hostPlatform.isWindows`, but `stdenv.hostPlatform.extensions.executable` is a better choice.

**Review by NickCao — COMMENTED** [review]
Commit message should match attribute name: `ooklaserver: init at 2.11.1.2`, otherwise LGTM.

File: pkgs/by-name/oo/ooklaserver/package.nix [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [ unzip autoPatchelfHook ];
+
+  installPhase = ''
+    runHook preInstall
+    unzip ${filenameMapping.${stdenv.system}}
+    install -Dm555 OoklaServer${stdenv.hostPlatform.extensions.executable} "$out/bin/OoklaServer"
+    install -Dm555 OoklaServer.properties.default "$out/etc/OoklaServer.properties.default"
```
Comment (NickCao) [inline pkgs/by-name/oo/ooklaserver/package.nix:40]:
```suggestion
    install -Dm444 OoklaServer.properties.default "$out/etc/OoklaServer.properties.default"
```

Oops missed the x bit.

**Review by NickCao — APPROVED** [review]
Diff LGTM, waiting for ofborg.

**Comment by NeverBehave** [comment]
Actually it might not need the `autoPatchelfHook`. Tried building without that and it is passing

**Comment by NickCao** [comment]
> Actually it might not need the `autoPatchelfHook`. Tried building without that and it is passing

Of course it builds, however do the resulting binary work?

**Comment by NeverBehave** [comment]
It is running fine and I double check it seems no deps is listed/required. I will remove this step for now.

**Comment by NickCao** [comment]
> It is running fine and I double check it seems no deps is listed/required. I will remove this step for now.

Ah it's a static binary, that's why.

**Comment by NeverBehave** [comment]
Updated without patch

**Comment by NeverBehave** [comment]
it looks like macOS evaluate failed because of unfree? 
https://logs.ofborg.org/?attempt_id=b6a345a9-ec98-4fbc-bb55-c5cce99a3e41&key=nixos%2Fnixpkgs.324410

**Pull request merged**